Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs

```diff
@@ -1,21 +1,35 @@
 using Content.Server.Chemistry.Components;
 using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Emag.Systems;
+using Content.Shared.Chemistry.Dispenser;
 using JetBrains.Annotations;
 using Robust.Shared.Containers;
-using Robust.Shared.GameObjects;
+using Robust.Shared.Prototypes;
 
 namespace Content.Server.Chemistry.EntitySystems
 {
     [UsedImplicitly]
     public sealed class ReagentDispenserSystem : SharedReagentDispenserSystem
     {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         public override void Initialize()
         {
             base.Initialize();
             SubscribeLocalEvent<ReagentDispenserComponent, ComponentStartup>((_, comp, _) => comp.UpdateUserInterface());
             SubscribeLocalEvent<ReagentDispenserComponent, SolutionChangedEvent>((_, comp, _) => comp.UpdateUserInterface());
             SubscribeLocalEvent<ReagentDispenserComponent, EntInsertedIntoContainerMessage>((_, comp, _) => comp.UpdateUserInterface());
             SubscribeLocalEvent<ReagentDispenserComponent, EntRemovedFromContainerMessage>((_, comp, _) => comp.UpdateUserInterface());
+            SubscribeLocalEvent<ReagentDispenserComponent, GotEmaggedEvent>(OnEmagged);
+        }
+
+        private void OnEmagged(EntityUid uid, ReagentDispenserComponent comp, GotEmaggedEvent args)
+        {
+            if (comp.AlreadyEmagged == false)
+            {
+                comp.InitializeFromPrototype(comp.EmagPackPrototypeId);
+                comp.AlreadyEmagged = true;
```

Comment:
shouldnt this 'already emagged' behavior be generalized

Reply:
like just an `EmaggedComponent` that gets added on a successful emag in the afterinteract, then don't raise a `GettingEmaggedEvent` if that comp exists

Reply:
this tbh, emag interactions are all very different and not generalizable so I want to let the other components decide everything

Reply:
alr thats fine